src/app/add-clients/helperFunctions.tsx

```diff
@@ -0,0 +1,254 @@
+    quantity?: number;
+}
+
+export interface ErrorMessages {
+    fullName: string;
+    phoneNumber: string;
+    addressLine1: string;
+    addressPostcode: string;
+    adults: string;
+    numberChildren: string;
+    nappyErrorMessage: string;
+}
+
+export interface Fields {
+    fullName: string;
+    phoneNumber: string;
+    addressLine1: string;
+    addressLine2: string;
+    addressTown: string;
+    addressCounty: string;
+    addressPostcode: string;
+    adults: People[];
+    numberChildren: number;
+    children: People[];
+    dietaryRequirements: booleanGroup;
+    feminineProducts: booleanGroup;
+    babyProducts: boolean | null;
+    nappySize: string;
+    petFood: booleanGroup;
+    otherItems: booleanGroup;
+    deliveryInstructions: string;
+    extraInformation: string;
+}
+
+export const setErrorFunction = (
+    errorSetter: React.Dispatch<React.SetStateAction<ErrorMessages>>,
+    errorValues: ErrorMessages
+): SetErrorType => {
+    return (errorKey, errorMessage): void => {
+        errorSetter({ ...errorValues, [errorKey]: errorMessage });
+    };
+};
+
+export const setFieldFunction = (
+    fieldSetter: React.Dispatch<React.SetStateAction<Fields>>,
+    fieldValues: Fields
+): SetFieldType => {
+    return (fieldKey, newFieldValue) => {
+        fieldSetter({ ...fieldValues, [fieldKey]: newFieldValue });
+    };
+};
+const getErrorType = (event: EventType, required?: boolean, regex?: RegExp): string => {
+    const input = event.target.value;
+    if (required && input === "") {
+        return "This is a required fields.";
+    } else if (!!regex && !input.match(regex)) {
+        return "Please enter a valid entry.";
+    } else {
+        return "";
+    }
+};
+
+export const checkboxGroupToArray = (checkedBoxes: booleanGroup): string[] => {
+    return Object.keys(checkedBoxes).filter((key) => checkedBoxes[key]);
+};
+
+export const onChangeFunction = (
+    fieldSetter: SetFieldType,
+    errorSetter: SetErrorType,
+    key: string,
+    required?: boolean,
+    regex?: RegExp,
+    formattingFunction?: (value: any) => any
+): OnChangeType => {
+    return (event) => {
+        const errorType = getErrorType(event, required, regex);
+        errorSetter(key, errorType);
+        if (errorType === "") {
+            const newValue = formattingFunction && formattingFunction(event.target.value);
+            fieldSetter(key, newValue);
+        }
+    };
+};
+
+export const onChangeCheckbox = (
+    fieldSetter: SetFieldType,
+    currentObject: booleanGroup,
+    key: string
+): OnChangeType => {
+    return (event) => {
+        const newObject = { ...currentObject, [event.target.name]: event.target.checked };
+        fieldSetter(key, newObject);
+    };
+};
+
+export const errorExists = (errorMessage: string): boolean => {
+    return errorMessage !== "" && errorMessage !== "N/A";
+};
+
+export const errorText = (errorMessage: string): string => {
+    return errorMessage == "N/A" ? "" : errorMessage;
```

Comment:
I reckon it makes more sense to encode this functionality into the type system instead of using a special value.